telemetry/statsd/src/main/scala/io/buoyant/telemetry/statsd/StatsDStatsReceiver.scala

```diff
@@ -28,21 +28,36 @@ private[telemetry] class StatsDStatsReceiver(
   }
   private[statsd] def close(): Unit = statsDClient.stop()
 
+  private[this] val counters = new ConcurrentHashMap[String, Metric.Counter].asScala
```

Comment:
might be more useful to keep these as ConcurrentHashMaps instead of `asScala`ing them so we get CHM's atomic methods